design/batch_retrieval_design.md

```diff
@@ -1,20 +1,354 @@
+| 3    | User/Host authenticated but secret doesn't exist             | Given I am a privileged, authenticated user/host<br />With proper `execute` permissions<br />Requesting secret does not exist | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/nonExistingSecret` in request body | - The JSON response should be (*see below #3*)<br />- Status response code should be `200 OK`<br />- Existing secret should have `200`. Non-existing secret should have `404 ` status error<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 4    | User/Host authenticated but secret does not have value       | Given I am a privileged, authenticated user/host<br />With proper permissions<br />Requesting secret *value* does not exist | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/nonSecretValue` in request body | - The JSON response should be (*see below #4)<br />- Status response code should be `200 OK`<br />- Existing secret should have `200`. Non-existing secret value should have `404 ` status error<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 5    | Invalid request body                                         | Given I am a privileged, authenticated user/host<br />With proper permissions<br />Requesting secret in invalid format | I send `POST /secrets` with invalid JSON body with `var/path/conjurSecret` and  `var/path/<>` | - The JSON response should be (see below #5)<br />- Status response code should be `400 `<br />- The proper audit record should appear for the whole request and each individual secret<br />- **@Elad should the whole request fail?** | Automated        |
+| 6    | Security Mitigation tests                                    |                                                              |                                                              |                                                              |                  |
+
+*Test 1 response body*
+
+```json
+200 OK
+
+{
+  "variables": [
+  {
+    "id": "var/path/conjurSecret",
+    "status": 200,
+    "value": "superSecret"
+  },
+  {
+    "id": "var/path/anotherConjurSecret",
+    "status": 200,
+    "value": "superSecret"
+  }
+ ] 
+}
+```
+
+*Test 2 response body*
+
+```json
+200 OK
+
+{
+  "variables": [
+  {
+    "id": "var/path/conjurSecret",
+    "status": 200,
+    "value": "superSecret"
+  },
+    {
+    "id": "var/path/anotherConjurSecret",
+    "status": 403,
+    "value": "superSecret"
+  },
+ ] 
+}
+```
+
+*Test 3 response body*
+
+```json
+200 OK
+
+{
+  "variables": [
+  {
+    "id": "var/path/conjurSecret",
+    "status": 200,
+    "value": "superSecret"
+  },
+    {
+    "id": "var/path/nonexistingSecret",
+    "status": 404,
```

Comment:
Should replace with :
"error": {
      "message": "Not Found"
 		}

Reply:
yes updating